python/pyarrow/tests/test_compute.py

```diff
@@ -1299,6 +1301,98 @@ def test_arithmetic_multiply():
+            expected_ = [3, 3, 3, 4, -3, -3, -3, None]
+        elif round_mode == "towards_infinity":
+            expected_ = [4, 4, 4, 5, -4, -4, -4, None]
+        elif round_mode == "half_down":
+            expected_ = [3, 3, 4, 4, -3, -4, -4, None]
+        elif round_mode == "half_up":
+            expected_ = [3, 4, 4, 5, -3, -3, -4, None]
+        elif round_mode == "half_towards_zero":
+            expected_ = [3, 3, 4, 4, -3, -3, -4, None]
+        elif round_mode == "half_towards_infinity":
+            expected_ = [3, 4, 4, 5, -3, -4, -4, None]
+        elif round_mode == "half_to_even":
+            expected_ = [3, 4, 4, 4, -3, -4, -4, None]
+        elif round_mode == "half_to_odd":
+            expected_ = [3, 3, 4, 5, -3, -3, -4, None]
+
+        expected = pa.array(expected_).to_numpy(zero_copy_only=False)
+        assert all(map(approx_equals, result, expected))
+
+
+def test_round():
+    approx_equals = partial(np.isclose, equal_nan=True)
+    round_ndigits = [-2, -1, 0, 1, 2]
+    arr = pa.array([320, 3.5, 3.075, 4.5, -3.212, -35.1234, -3.045, None])
+    for ndigits in round_ndigits:
+        options = pc.RoundOptions(
+            ndigits=ndigits, round_mode="half_towards_infinity")
+        result = pc.round(arr, options=options).to_numpy(zero_copy_only=False)
+        if ndigits == -2:
+            expected_ = [300, 0, 0, 0, 0, 0, 0, None]
+        elif ndigits == -1:
+            expected_ = [320, 0, 0, 0, 0, -40, 0, None]
+        elif ndigits == 0:
+            expected_ = [320, 4, 3, 5, -3, -35, -3, None]
+        elif ndigits == 1:
+            expected_ = [320, 3.5, 3.1, 4.5, -3.2, -35.1, -3, None]
+        elif ndigits == 2:
+            expected_ = [320, 3.5, 3.08, 4.5, -3.21, -35.12, -3.05, None]
+
+        expected = pa.array(expected_).to_numpy(zero_copy_only=False)
+        assert all(map(approx_equals, result, expected))
+
+
+def test_mround():
+    approx_equals = partial(np.isclose, equal_nan=True)
+    multiples = [-2, -0.05, 0.1, 0, 10, 100]
+    arr = pa.array([320, 3.5, 3.075, 4.5, -3.212, -35.1234, -3.045, None])
+    for multiple in multiples:
+        options = pc.MRoundOptions(
+            multiple=multiple, round_mode="half_towards_infinity")
+        result = pc.mround(arr, options=options).to_numpy(zero_copy_only=False)
+        if multiple == -2:
+            expected_ = [320, 4, 4, 4, -4, -36, -4, None]
+        elif multiple == -0.05:
+            expected_ = [320, 3.5, 3.1, 4.5, -3.2, -35.1, -3.05, None]
+        elif multiple == 0.1:
+            expected_ = [320, 3.5, 3.1, 4.5, -3.2, -35.1, -3, None]
+        elif multiple == 0:
+            expected_ = [0, 0, 0, 0, 0, 0, 0, None]
+        elif multiple == 10:
+            expected_ = [320, 0, 0, 0, 0, -40, 0, None]
+        elif multiple == 100:
+            expected_ = [300, 0, 0, 0, 0, 0, 0, None]
+
+        expected = pa.array(expected_).to_numpy(zero_copy_only=False)
+        assert all(map(approx_equals, result, expected))
```

Comment:
nit: it would be more natural to use np.all(np.isclose(arr1, arr2, equal_nan=True)) instead of using map/partial/all.